In [15]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns 
import warnings
warnings.filterwarnings('ignore')

In [16]:
df=pd.read_csv('E://desktop/data/Salary_Data.csv')
df

YearsExperience    Salary
0               1.1   39343.0
1               1.3   46205.0
2               1.5   37731.0
3               2.0   43525.0
4               2.2   39891.0
5               2.9   56642.0
6               3.0   60150.0
7               3.2   54445.0
8               3.2   64445.0
9               3.7   57189.0
10              3.9   63218.0
11              4.0   55794.0
12              4.0   56957.0
13              4.1   57081.0
14              4.5   61111.0
15              4.9   67938.0
16              5.1   66029.0
17              5.3   83088.0
18              5.9   81363.0
19              6.0   93940.0
20              6.8   91738.0
21              7.1   98273.0
22              7.9  101302.0
23              8.2  113812.0
24              8.7  109431.0
25              9.0  105582.0
26              9.5  116969.0
27              9.6  112635.0
28             10.3  122391.0
29             10.5  121872.0

In [17]:
from sklearn.model_selection import train_test_split
xtrain,xtest,ytrain,ytest=train_test_split(df['YearsExperience'],df['Salary'], test_size=0.30)
print(xtrain, xtest)

5      2.9
8      3.2
7      3.2
28    10.3
24     8.7
2      1.5
13     4.1
11     4.0
3      2.0
19     6.0
23     8.2
27     9.6
1      1.3
14     4.5
17     5.3
26     9.5
6      3.0
25     9.0
0      1.1
10     3.9
16     5.1
Name: YearsExperience, dtype: float64 4      2.2
29    10.5
9      3.7
12     4.0
15     4.9
20     6.8
18     5.9
21     7.1
22     7.9
Name: YearsExperience, dtype: float64


In [19]:
xtrain=np.array(xtrain).reshape(-1,1)
xtest=np.array(xtest).reshape(-1,1)
ytrain=np.array(ytrain).reshape(-1,1)
ytest=np.array(ytest).reshape(-1,1)
from sklearn.linear_model import LinearRegression
lr=LinearRegression()
lr.fit(xtrain,ytrain)
lr.score(xtrain, ytrain)


0.9530082519864688

#r2 score is 95%, model is predicting 95% correct values

In [20]:
lr.predict(xtest)

array([[ 47861.00000533],
       [124861.5614517 ],
       [ 61776.76412214],
       [ 64559.9169455 ],
       [ 72909.37541559],
       [ 90536.00996356],
       [ 82186.55149347],
       [ 93319.16278692],
       [100740.90364922]])

In [22]:
ytest

array([[ 39891.],
       [121872.],
       [ 57189.],
       [ 56957.],
       [ 67938.],
       [ 91738.],
       [ 81363.],
       [ 98273.],
       [101302.]])

#problem is we have candidate with 5 years of experience and we want to predict best salary that company can offer him

In [26]:
lr.predict([[5]])

array([[73837.09302338]])

#so, our model is predicting salary of 73837.09302338, so company can offer that person salary in range of 70-75